pkg/applyinator/file_test.go

```diff
@@ -0,0 +1,300 @@
+//go:build !arm64
+
+package applyinator
+
+import (
+	"encoding/base64"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+func TestWriteContentToFile(t *testing.T) {
+	tempDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	getFile := func(path string, permissions string, content string) File {
+		return File{
+			Permissions: permissions,
+			Path:        filepath.Join(tempDir, path),
+			Content:     content,
+		}
+	}
+
+	testCases := []struct {
+		Path        string
+		Permissions string
+		Content     string
+
+		Base64Encode bool
+
+		ExpectedPermissions os.FileMode
+		ExpectedErr         bool
+	}{
+		{
+			Path:    "test-no-perms",
+			Content: "hello world",
+
+			ExpectedPermissions: defaultFilePermissions,
+			ExpectedErr:         false,
+		},
+		{
+			Path:        "test-perms",
+			Permissions: "0600",
```

Comment:
it seems that `defaultFilePermissions` is also 0600, could we change the value here, and in other cases, to a different one?

Reply:
I've updated the perms in `TestWriteContentToFile` to `0666`